src/config/mod.rs

```diff
@@ -206,22 +207,33 @@ pub struct FileDialogConfig {
 }
 
 impl Default for FileDialogConfig {
-    /// Creates a new configuration with default values
     fn default() -> Self {
+        Self::default_from_filesystem(&NativeFileSystem)
+    }
+}
+
+impl FileDialogConfig {
+    /// Creates a new configuration with default values
+    pub fn default_from_filesystem(fs: &dyn FileSystem) -> Self {
```

Comment:
Could we rename this to `default_with_filesystem`, so it matches with `with_config`?